### DISCUSSION TITLE
[4.x] Eager loaded elements are null

### DISCUSSION THREAD

**Discussion by martinherweg** [craftcms/cms#12247] [Q&A] [answered]
Hey,

not sure if this is a bug or an issue on our side so a discussion rather then an issue.

We have quite a complicated page with some filtering, relations etc. where we have a problem with eager loading of an entries field. 

Even if I simplify the query the problem is the same.

```twig
{% set newsQuery = craft.entries.section('news').with(['heroImage', 'topics', 'category']).limit(4).all() %}
{% for news in newsQuery %}
  {{ d(news.topics.one()) }}
{% endfor %}
```

The result of the dump is `null` for all elements, the topics are Entries and are surely set on these elements and the category for example works fine when eager loaded.
If I remove `topics` from the `.with()` I get results as wished.

Any hint on what might be the problem here and what I could look at?

Thanks.

EDIT:
Additional Info would be that it works in a different environment just fine.

**Comment by brianjhanson** [comment, marked as answer]
Edit because I was wrong. 

Since Craft 4, eager loaded elements are returned as an `craft\elements\ElementCollection` which doesn't care if you use `[0]` or `.one()`.

The (mostly) original answer below still applies to Craft 3 though. 

You'll need to use the array syntax when [accessing eager loaded elements](https://craftcms.com/docs/4.x/dev/eager-loading-elements.html#accessing-eager-loaded-elements). 

```
{% set newsQuery = craft.entries.section('news').with(['heroImage', 'topics', 'category']).limit(4).all() %}
{% for news in newsQuery %}
  {{ d(news.topics[0]) }}
{% endfor %}
```

**Reply by martinherweg** [reply]
Hey @brianjhanson,

that is not correct, eager loaded elements are always a collection, your way only the newsQuery itself would also be collection.
Array syntax even throws an error `Key "0" in object with ArrayAccess of class "craft\elements\ElementCollection" does not exist.`

**Reply by brianjhanson** [reply]
Whoops you're right, I'll update my answer. Either way, you should still be able to access `news.topics` with either `[0]` or `.one()`. Can you open an issue for this so we can check it out? Any additional detail you can provide would be helpful as I'm not able to replicate with a simple project.
